matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/SendingEventsDataSource.kt

```diff
@@ -72,7 +72,8 @@ internal class RealmSendingEventsDataSource(
         if (frozenSendingTimelineEvents?.isValid == true) {
             frozenSendingTimelineEvents?.realm?.close()
         }
-        frozenSendingTimelineEvents = sendingEvents?.freeze()
+        // Do not freeze empty list
+        frozenSendingTimelineEvents = sendingEvents?.takeIf { it.isNotEmpty() }?.freeze()
```

Comment:
Maybe we could map the entities with timelineEventMapper, and keep that in memory. Instead of a list of frozen.
We could then remove the freeze()